### PR TITLE
Addition of basic auth support

### PR DESCRIPTION
**Pull request by matthewsullivan-wf**
Following the pattern defined in https://github.com/splunk/splunk-sdk-ruby/pull/43, I have implemented Basic Auth in this PR.

```
from splunklib.client import Service

service = Service(host="splunk.myhost.net", port=8089, scheme="https", version=6.3,
                  basic=True, username="steve", password="puppydoge")

service.indexes['test'].submit("My message!", host="localhost", sourcetype="text")
```

### CODE REVIEW

**Comment by matthewsullivan-wf** [comment]
@itay any idea what I'm doing wrong getting this test set up?
https://travis-ci.org/splunk/splunk-sdk-python/jobs/98574591#L602

**Comment by matthewsullivan-wf** [comment]
@itay ^

**Comment by itay** [comment]
Hey @matthewsullivan-wf - sorry it took so long to reply. I finally had some time to look today.

Anyway, I found the issue with your test. I put a comment on the specific line, but I wanted to ask you how you wanted to proceed. I can merge your PR once that's fixed (and if you want to rebase it to the tip of develop and squash your commits), or I can just fix it and get that merged. I wanted to give you the option of doing it yourself so you get "credit" for the contribution on GitHub, since you did the hard work :)

Let me know how you want to proceed.

File: tests/test_binding.py [hunk 1]
```diff
@@ -686,6 +686,32 @@ def test_namespace(self):
     def test_namespace_fails(self):
         self.assertRaises(ValueError, binding.namespace, sharing="gobble")
 
+class TestBasicAuthentication(unittest.TestCase):
+    def setUp(self):
+        self.opts = testlib.parse([], {}, ".splunkrc")
+        opts = self.opts.kwargs.copy()
+        opts["basic"] = True
+        opts["username"] = "boris the mad baboon"
```
Comment (itay) [inline tests/test_binding.py]: You're trying to log in with a bad set of credentials. Change this line to: 

```
        opts["username"] = self.opts.kwargs["username"]
        opts["password"] = self.opts.kwargs["password"]
```

**Comment by sullivanmatt** [comment]
@itay appreciate the consideration for credit, but I don't care - I just need it fixed so we can start using the SDK at my place of work :)  Go ahead if you've got everything staged that you need.

**Comment by itay** [comment]
Superceded by #133

**Pull request closed without merging**